cmd/juju/machine/add.go

```diff
@@ -168,7 +169,47 @@ type MachineManagerAPI interface {
 	Close() error
 }
 
-var manualProvisioner = manual.ProvisionMachine
+// manualProvisioner provisions a machine agent to an existing host, via
+// an SSH connection or WinRM connection to the specified host. The host may optionally be preceded
+// with a login username, as in [user@]host.
+//
+// On successful completion, this function will return the id of the state.Machine
+// that was entered into state.
+//
+// we use manualProvisioner as a lambda and not a function
+// because this variable will be overwritten in tests.
+var manualProvisioner = func(args manual.ProvisionMachineArgs,
```

Comment:
I think all of this should be in the "scopeRun" function. We should only be mocking at the boundaries, i.e. manual.ProvisionMachine/sshprovisioner.New.